### PR TITLE
Removal of es-module-shims dependency

### PR DESCRIPTION
**Pull request by jubalm**
Most modern browsers now support importmap except Safari. Open for discussion
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap

- [ ] added prettier config to prevent rewriting current code structure

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -1,15 +1,13 @@
 <!DOCTYPE html>
 
 <head>
-	<meta charset='utf-8'>
-	<link rel='stylesheet' href='./css/index.css'>
+	<meta charset="utf-8" />
+	<link rel="stylesheet" href="./css/index.css" />
```
Comment (MicahZoltu) [inline app/index.html:5]: Why did these get switched to double quotes?  Is the intent to change the style, or was this an accident?  IF it was intentional, it should be in a separate PR.

File: app/index.html [hunk 2]
```diff
@@ -20,7 +18,7 @@
 		}
 	</script>
 
-	<main>Loading...</main>
+	<main></main>
```
Comment (MicahZoltu) [inline app/index.html:21]: Recommend retaining this `Loading...`.  In case the script loads slowly, this lets the user know that something is happening while they wait for Preact's first render.

File: build/vendor.ts [hunk 3]
```diff
@@ -21,12 +23,14 @@ async function vendorDependencies() {
...
-	const importmapJson = JSON.stringify(importmap, undefined, '\t')
-		.replace(/^/mg, '\t\t')
+	const importmap = dependencyPaths.reduce(
+		(importmap, { packageName, entrypointFile }) => {
+			importmap.imports[packageName] = `./${path.join('.', 'vendor', packageName, entrypointFile).replace(/\\/g, '/')}`
+			return importmap
+		},
+		{ imports: {} as Record<string, string> }
+	)
+	const importmapJson = JSON.stringify(importmap, undefined, '\t').replace(/^/gm, '\t\t')
```
Comment (MicahZoltu) [inline build/vendor.ts:33]: Why was this changed?  Removal of `es-module-shims` shouldn't have any impact on this file other than removing `es-module-shims` as a dependency.

File: app/index.html [hunk 4]
```diff
@@ -20,7 +18,7 @@
...
 
-	<script type='module' src='./js/index.js'></script>
+	<script type="module" src="./js/index.js"></script>
```
Comment (MicahZoltu) [inline app/index.html:23]:
```suggestion
	<script async type="module" src="./js/index.js"></script>
```
While not required, we recently learned making this `async` is probably more correct and may help avoid some weird race conditions in FF.

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
There are some changes in this PR that seem unrelated to removing `es-module-shims` that should be either removed (if unintentional) or moved to a different PR.

File: app/index.html (hunk 1 above)
Comment (jubalm) [reply app/index.html:5]: the editor automatically formats it to double quotes when saving even if prettier is already set to single quotes. amended

File: build/vendor.ts (hunk 3 above)
Comment (jubalm) [reply build/vendor.ts]: the editor formats them on save. reverting these lines to avoid confusion

File: app/index.html (hunk 1 above)
Comment (MicahZoltu) [reply app/index.html]: Hmm, this suggests you have some editor setting configured.  I am able to save `.html` files in VSCode without it converting `'` to `"` for me.

File: app/index.html (hunk 2 above)
Comment (jubalm) [reply app/index.html]: Was going to build a fancier one but yeah, will move it to another PR anyways so reverted for now

**Review by MicahZoltu — APPROVED** [review]
